### ISSUE TITLE
Set inventory quantity to a big number may cause out of memory error

### ISSUE THREAD

**Issue by duanhong169** [reactioncommerce/reaction#3628]
### Expected behavior
Can publish correctly.

### Actual behavior
Server crashed.

### Steps to reproduce the behavior
1. Login as shop owner
2. Edit a product variant
3. Set inventory quantity to a big number(e.g. 1000000000)
4. Click 'Publish'

### Versions
1.7.0

### Related code

https://github.com/reactioncommerce/reaction/blob/f40ff536c139d70da02ca10ae12655247452d658/imports/plugins/included/inventory/server/methods/inventory.js#L52-L66

### Crash log

```
Feb 01 14:53:41 iZrj90nq43piiv539ao8fyZ xxxx[7775]: <--- Last few GCs --->
Feb 01 14:53:41 iZrj90nq43piiv539ao8fyZ xxxx[7775]: [7775:0x26b9430]   951501 ms: Mark-sweep 1400.8 (1473.4) -> 1400.8 (1473.4) MB, 1364.2 / 0.0 ms  allocation failure GC in old space requested
Feb 01 14:53:41 iZrj90nq43piiv539ao8fyZ xxxx[7775]: [7775:0x26b9430]   952965 ms: Mark-sweep 1400.8 (1473.4) -> 1400.7 (1443.4) MB, 1461.5 / 0.0 ms  last resort GC in old space requested
Feb 01 14:53:41 iZrj90nq43piiv539ao8fyZ xxxx[7775]: [7775:0x26b9430]   954530 ms: Mark-sweep 1400.7 (1443.4) -> 1400.7 (1443.4) MB, 1564.5 / 0.0 ms  last resort GC in old space requested
Feb 01 14:53:41 iZrj90nq43piiv539ao8fyZ xxxx[7775]: <--- JS stacktrace --->
Feb 01 14:53:41 iZrj90nq43piiv539ao8fyZ xxxx[7775]: ==== JS stack trace =========================================
Feb 01 14:53:41 iZrj90nq43piiv539ao8fyZ xxxx[7775]: Security context: 0x2f5c4bf25e91 <JSObject>
Feb 01 14:53:41 iZrj90nq43piiv539ao8fyZ xxxx[7775]: 1: _registerInventory(aka _registerInventory) [/home/remote/build/bundle/programs/server/app/app.js:~54457] [pc=0x30e0a3dcb8d5](this=0x3f1999a02311 <undefined>,product=0x11be82d2bce9 <Object map = 0xd3133365049>)
Feb 01 14:53:41 iZrj90nq43piiv539ao8fyZ xxxx[7775]: 3: inventoryRegister [/home/remote/build/bundle/programs/server/app/app.js:54679] [bytecode=0x11be82d2be01 offset=67](this=0x17788884d001 <JSObject>,product=0x11be82d2bce9 <Obje...
Feb 01 14:53:41 iZrj90nq43piiv539ao8fyZ xxxx[7775]: FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
Feb 01 14:53:41 iZrj90nq43piiv539ao8fyZ xxxx[7775]: 1: node::Abort() [/usr/local/bin/node]
Feb 01 14:53:41 iZrj90nq43piiv539ao8fyZ xxxx[7775]: 2: 0x121809c [/usr/local/bin/node]
Feb 01 14:53:41 iZrj90nq43piiv539ao8fyZ xxxx[7775]: 3: v8::Utils::ReportOOMFailure(char const*, bool) [/usr/local/bin/node]
Feb 01 14:53:41 iZrj90nq43piiv539ao8fyZ xxxx[7775]: 4: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [/usr/local/bin/node]
Feb 01 14:53:41 iZrj90nq43piiv539ao8fyZ xxxx[7775]: 5: v8::internal::Factory::NewFillerObject(int, bool, v8::internal::AllocationSpace) [/usr/local/bin/node]
Feb 01 14:53:41 iZrj90nq43piiv539ao8fyZ xxxx[7775]: 6: v8::internal::Runtime_AllocateInTargetSpace(int, v8::internal::Object**, v8::internal::Isolate*) [/usr/local/bin/node]
Feb 01 14:53:41 iZrj90nq43piiv539ao8fyZ xxxx[7775]: 7: 0x30e0a390463d
```

**Comment by prinzdezibel** [comment]
Talked to Brent about this, mainly to get an idea why every instance of a product get its own document in db.  Under the assumption that we really want to keep this as it is (i.e. one document for each instance), I would propose to arbitrarily constrain the highest inventory amount to say "1000000".  Not sure if there's a valid use case to allow values of 1 billion or higher. But that's just a first proposal, maybe others have better ideas.

**Comment by zenweasel** [comment]
Since the use case for this is pretty limited, I would consider this a low-impact bug and thus a lower priority

**Comment by aldeed** [comment]
Fixed in 2.0.0
